files/os_patching_fact_generation.sh

```diff
@@ -17,7 +17,17 @@ else
   echo "$$" > $LOCKFILE
 fi
 
-case $(facter osfamily) in
+if [ $(puppet --version|cut -d. -f1) -gt 6 ]; then
+  OSFAMILY=$(puppet facts show --render-as s osfamily | cut -d\" -f4)
+  VARDIR=$(puppet facts show --render-as s puppet_vardir | cut -d\" -f4)
```

Comment:
That seems like it should be a `puppet config print vardir` ?

Reply:
> That seems like it should be a `puppet config print vardir` ?

Thanks. I am gonna change it. It's less consistent in the whole script, but it's quicker to execute.

Reply:
I saw both methods in there too... thats where I got the idea, thinking "vardir" really sounds more like a config than a fact. I am pretty sure osfamily and stuff will still need to be gathered as facts. Another "point of interest" is that I think "osfamily" is a *legacy* fact that was deprecated? Perhaps we should go with os.family (structured fact)? (and I am sure there are others like that?)